Orm/Xtensive.Orm/Orm/Rse/Providers/Provider.cs

```diff
@@ -136,7 +127,7 @@ protected virtual string ParametersToString()
     /// </summary>
     /// <param name="type">The type of the provider.</param>
     /// <param name="sources"><see cref="Sources"/> property value.</param>
-    protected Provider(ProviderType type, params Provider[] sources)
+    protected Provider(ProviderType type, IReadOnlyList<Provider> sources)
```

Comment:
Can you explain what was the problem here? Native array is better, isn't it?
Was it for safety reasons so user code wouldn't be able to change items?

This is a breaking change which generates a breaking change of the ```ExecutableProvider``` ctor, so I need to evaluate its pros.

Reply:
The only reason: make `Provider.Sources` readonly collection